packages/app-project/src/screens/ClassifyPage/components/ClassifierWrapper/ClassifierWrapper.js

```diff
@@ -83,11 +83,12 @@ export default function ClassifierWrapper({
   /*
     Track the current classification subject, when it changes inside the classifier.
   */
-  const onSubjectChange = useCallback((subject) => {
-    if (subjectInURL) {
-      const subjectPageURL = `${baseURL}/subject/${subject.id}`
-      const href = addQueryParams(subjectPageURL)
-      replaceRoute(href, href, { shallow: true })
+  const onSubjectChange = useCallback(
+    subject => {
+      if (subjectInURL) {
+        const subjectPageURL = `${baseURL}/subject/${subject.id}`
+        const href = addQueryParams(subjectPageURL)
+        replaceRoute(href, href, { scroll: false })
```

Comment:
Won't this reinitialise the classifier and run data-fetching each time the URL changes?

Reply:
It will also run `getStaticProps` every time you submit a classification.
https://nextjs.org/docs/pages/building-your-application/routing/linking-and-navigating#shallow-routing

Reply:
Yep, pressing Next or Previous on this branch re-initialises the classifier and requests a new subject queue from Panoptes. I didn't try submitting a classification, but I think pressing Done will also reset the subject queue.

The problem here is that we want to update the page URL in the window location, but we don't actually want the Next.js app to navigate to a new Classify page. Navigating to a new page will restart the classifier. Hence the shallow route change in the original code.

Reply:
@eatyourgreens Do you have a suggestion solution for this bug fix? (other than full upgrade to Next 14).

I think you have a deeper understanding of your implementation of subject set selection, prioritized subjects, etc.

Reply:
Maybe avoid the Next.js router entirely and use [`history.replaceState`](https://developer.mozilla.org/en-US/docs/Web/API/History/replaceState)?

Reply:
I can also reproduce the empty `query` bug by going directly to a subject page:
https://www.zooniverse.org/projects/printmigrationnetwork/print/classify/workflow/20199/subject-set/111066/subject/84319199

But I don’t see that bug on Gravity Spy if I go directly to a workflow, so that’s odd. I’d expect all Classify pages to have the bug. 
https://www.zooniverse.org/projects/zooniverse/gravity-spy/classify/workflow/1610

Reply:
The empy `query` bug only occurs when a subject is in the URL. It's caused by line 91 above: `replaceRoute(href, href, { scroll: false })`. When that line is removed, the router has expected `query` keys.

Reply:
Or I should be clear, the empty `query` bug only occurs when `router.replace` uses shallow routing. When I was debugging and removed that code line, the router isn't emptied.

Reply:
So removing that line fixes the broken links to the home page and to the project stats? 🤔

Reply:
Yes, removing `router.replace` restores the broken links. I haven't tried `history.replaceState` yet, but will give that a go today.

Reply:
That’s useful to know, because this callback shouldn’t be running during a page load.

Reply:
@eatyourgreens what would be the downsides of re-initializing the classifier and requesting a new subject queue every time a volunteer uses the Next/Prev buttons for prioritized subjects? I ask because [shallow routing is not supported](https://github.com/vercel/next.js/discussions/48110) in NextJS 13 👎 Relying on `router.replace` is a blocker for the Next/Prev feature going forward. I think I understanding the motivation behind using shallow routing for this feature originally, but now there's a disconnect between wanting the subject id reflected in the url, controlling the url via NextJS, yet controlling the current subject id in the isolated Classifier component. I would rather the url be the source of truth, and have the Classifier component react.

`history.replaceState()` is not a solution because modifying the url via `replaceState` or `pushState` does not preserve app state, and the browser back button breaks.

Reply:
>@eatyourgreens what would be the downsides of re-initializing the classifier and requesting a new subject queue every time a volunteer uses the Next/Prev buttons for prioritized subjects? 

This change navigates the Next.js app to a new Classify page on each subject change (so whenever Next, Previous or Done are clicked.) Think about what the implications are for building a new Classify page for each new subject, what this implies for the Panoptes API, and what impact that might have on the backend with, say, ~100 volunteers classifying across ~10 transcription projects.